Assets/Scripts/UI/Canvas/StateMachine/UIStateManager.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using UI.Canvas.Introduction;
+using UI.Canvas.PhotoBookSystem;
+using UnityEngine;
+
+namespace UI.Canvas.StateMachine
+{
+    public class UIStateManager : MonoBehaviour
+    {
+        private IntroductionPopup _introductionPopup;
+        private PhotoBook _photoBook;
+        private void Awake()
+        {
+            _introductionPopup = FindObjectOfType<IntroductionPopup>();
+            _photoBook = FindObjectOfType<PhotoBook>();
+        }
```

Comment:
Zelf denk ik dat dit niet nodig is, er zijn maar 5 UI states en die hebben altijd het zelfde object. Hier kan makkelijk een standaard refrence gemaakt worden.